### ISSUE TITLE
[Help wanted] Translate the four action blocks that are still English in 12 languages

### ISSUE THREAD

**Issue by bvis** [bvis/aegis-hass#475]
Four action blocks are present in every language file but still hold their English text. The keys are all there — a test now enforces that — so this is purely about the values, and it is the kind of thing a native speaker fixes in two minutes and a maintainer guesses badly.

**Status (updated 2026-09-09).** Done: `it` (wip3out3r, #477), `nl` (aavdberg, #483 + #484), `es` (#485), `uk` — which already carried all 26 values in Cyrillic — and `da`, a whole new language contributed by webliodk in #488 with exact key parity. `de`, `fr`, `pt`, `pt-BR` and `ca` are in #493, translated by the maintainer from each file's existing vocabulary (see the status comment below for why, and what still needs a native eye there). **Four languages left, each needing a native speaker: `cs`, `pl`, `ro`, `tr`.** #493 also localises the `press_panic_button` field names (`Entity` / `Latitude` / `Longitude`), a fifth gap that was still English in 13 of 14 locales.

~~Affected in 12 of the 13 non-English languages (`es` already has them):~~ Correction: `es` did **not** already have them. Only `set_photo_on_demand_mode` was translated; the three session blocks — including both irreversible sign-out confirmations — were still English, which is the worst place in the file to be wrong. Found while auditing what the Dutch PR left, and fixed in #485.

| block | what a user sees today |
|---|---|
| `services.list_client_sessions` | name, description and the `entry_id` field, in English |
| `services.terminate_client_session` | name, description, all three field names and descriptions, in English |
| `services.terminate_other_client_sessions` | name, description, both field names and descriptions, in English |
| `services.set_photo_on_demand_mode` | name, description and the `entity_id` / `scenario` / `user` fields, in English |

They appear in the Home Assistant action picker and in the action's own form, so a Polish or Turkish user configuring an automation reads English there while the rest of the integration is in their language.

**Why this is being asked for rather than done.** Two of these actions sign devices out of the Ajax account and cannot be undone, and their `confirm` field is the text that tells you what you are about to do. Machine-translating a destructive confirmation into a language the maintainer cannot verify is a worse outcome than the English that is already there — so those values were deliberately left alone, and are tracked here instead.

**How to help.** Pick your language, open `custom_components/aegis_ajax/translations/<lang>.json`, and translate the values of the blocks above; leave the keys exactly as they are and ~~leave `FCM Project ID` / `FCM App ID` / `FCM API Key` / `FCM Sender ID` in English, since those are the labels Google itself uses~~ **translate the FCM labels like any other string** — this instruction was wrong and all 14 existing locales already ignore it (`Clave de API FCM`, `FCM-API-Schlüssel`, `Ключ API FCM`). Keep the `FCM` initialism itself, translate the rest. A pull request touching only your own language file is perfect — no need to wait for anyone else's, and no need to do all four blocks if you only want to do one. If you would rather not open a PR, pasting the translated block in a comment here works just as well.

The same offer stands for anything else in your language that reads badly. The existing translations were not written by native speakers of most of these languages, and a few carry ASCII transliterations of their own alphabets — `uk` had a Latin-transliterated repair dialog until #474 — so corrections are welcome beyond the list above.

**Comment by wip3out3r** [comment]
Italian (`it`), all four blocks. I am a native speaker and this is my own install's language, so these are the strings I actually read in the action picker.

Thank you for leaving these in English rather than machine-translating them: the `confirm` field on the two sign-out actions is exactly the text that tells someone what they are about to do, and guessing at it in a language you cannot check would have been the worse outcome.

I took the wording from the existing `it.json` rather than inventing it, so the four blocks match what is already there: *space* is `spazio`, *device* is `dispositivo`, *entity* is `Entità`, *session* is `sessione`, *service* is `servizio`, *Photo on Demand* is `Foto su richiesta`, and *hub* and *account* stay as they are. `Foto su richiesta` is also what the Ajax app itself says in Italian: under *Privacy* the two toggles read `Consenti foto su richiesta` and `Consenti foto per scenari`, and the second one spells out *la funzione Foto su richiesta*. Those two toggles line up with the two channels this action sets, so the wording matches what the user already reads in the app. `Confirm termination` follows the existing `Confirm panic` -> `Conferma antipanico`, which is why it has no article. Of the four action names already translated in the file two are imperative (`Disarm night mode` -> `Disinserisci modalità notte`, `Press panic button` -> `Premi il pulsante antipanico`), so `Elenca...`, `Disconnetti...` and `Imposta...` follow that precedent rather than introducing a new style.

One choice in here is mine rather than a copy, and I would rather flag it than hide it. **`Terminate`** has no precedent in the file: the only translated line that uses the word is `fcm_push_stuck`, about a connection ending, which is a different sense. I rendered it `Disconnetti` / `Conferma disconnessione` rather than `Termina` / `Conferma terminazione`, because in Italian `terminare una sessione` reads closer to killing a process than to signing a device out. If you would rather have `Termina`, say so and I will swap it everywhere in the block.

**`Config entry ID`** does not appear anywhere else in `it.json`, so the file dictates nothing there either, but I did not have to invent it: Home Assistant core ships that exact string in Italian as `ID voce di configurazione`, on `homeassistant.reload_config_entry`, and that is what I used.

That is 26 strings across the four blocks, counted against `translations/it.json` on `main`, where all four are still English. Five of them are field descriptions that exist in `en.json` on `main` but not in the `1.19.0-beta.6` I run, so those I translated from `main` rather than from my install. If the files move again before you take this, the newer file wins over this comment.

```json
"list_client_sessions": {
  "name": "Elenca le sessioni dell'account",
  "description": "Restituisce su richiesta le sessioni attive dell'account Ajax, per un'ispezione manuale. Questo servizio chiama un endpoint interno di Ajax che può limitare la frequenza delle richieste: non interrogarlo ciclicamente nelle automazioni. La sessione Aegis attualmente attiva è contrassegnata con is_current quando è identificabile.",
  "fields": {
    "entry_id": {
      "name": "ID voce di configurazione",
      "description": "Necessario quando è configurato più di un account Aegis."
    }
  }
},
"terminate_client_session": {
  "name": "Disconnetti una sessione dell'account",
  "description": "Disconnette immediatamente dall'account Ajax il dispositivo selezionato. Ottieni il suo ID sessione con list_client_sessions. L'integrazione non disconnette mai la propria sessione.",
  "fields": {
    "session_id": {
      "name": "ID sessione",
      "description": "ID sessione restituito da list_client_sessions."
    },
    "confirm": {
      "name": "Conferma disconnessione",
      "description": "Deve essere true per disconnettere immediatamente la sessione selezionata."
    },
    "entry_id": {
      "name": "ID voce di configurazione",
      "description": "Necessario quando è configurato più di un account Aegis."
    }
  }
},
"terminate_other_client_sessions": {
  "name": "Disconnetti tutte le altre sessioni dell'account",
  "description": "Disconnette immediatamente dall'account Ajax tutti gli altri dispositivi. Fa uscire le altre app Ajax mobili e desktop attive. L'integrazione non disconnette mai la propria sessione.",
  "fields": {
    "confirm": {
      "name": "Conferma disconnessione",
      "description": "Deve essere true per disconnettere immediatamente tutte le altre sessioni."
    },
    "entry_id": {
      "name": "ID voce di configurazione",
      "description": "Necessario quando è configurato più di un account Aegis."
    }
  }
},
"set_photo_on_demand_mode": {
  "name": "Imposta la modalità Foto su richiesta",
  "description": "Attiva o disattiva la modalità Foto su richiesta su un hub. Si possono commutare separatamente due canali indipendenti: le richieste degli utenti (se gli utenti dell'hub possono richiedere foto dall'app Ajax) e gli scenari (se scenari e automazioni possono avviare gli scatti). Lascia un campo non impostato per mantenerne il valore attuale. Almeno uno è obbligatorio.",
  "fields": {
    "user": {
      "name": "Richieste degli utenti",
      "description": "Se impostato, attiva o disattiva le richieste di foto avviate dagli utenti dell'hub. Lascia non impostato per mantenere il valore attuale."
    },
    "scenario": {
      "name": "Scenari",
      "description": "Se impostato, attiva o disattiva gli scatti Foto su richiesta avviati da scenari o automazioni Ajax. Lascia non impostato per mantenere il valore attuale."
    },
    "entity_id": {
      "name": "Entità",
      "description": "Pannello allarme dello spazio di cui aggiornare la modalità dell'hub. Se omesso, si applica a tutti gli spazi configurati."
    }
  }
}
```

Pasting it here rather than opening a PR, as you offered. Say the word if a PR is easier to merge and I will send one, and I am happy to look at anything else in Italian that reads wrong.

**Comment by bvis** [comment]
Taken as written, in #477 — thank you. This is the version of this contribution I was hoping for and did not really expect: not 26 translated strings, but 26 strings plus the reason for each one.

Three things in your note did work I could not have done:

- **`Foto su richiesta` is the app's own Italian.** You checked what Ajax says under *Privacy* — `Consenti foto su richiesta` and `Consenti foto per scenari` — and those two toggles are exactly the two channels this action sets. So the action now uses the words the user already reads in the app, which is better than a correct translation.
- **`Terminate` → `Disconnetti`, and you flagged it as yours.** Keeping it. You are the person who has to read that string, and *terminare una sessione* sounding like killing a process is precisely the kind of thing that does not survive a dictionary. Flagging the one invented choice instead of burying it among the copied ones is what made this reviewable at all.
- **`ID voce di configurazione` from Home Assistant core.** Matching core's own string for the same concept is the right instinct, and it is not one I would have thought to check.

Also noted, and correct: five of the field descriptions exist on `main` but not in the `1.19.0-beta.6` you run. You translated from `main`, which is what I would have asked for. `1.19.0` is out now, so your install and the file agree again.

Leaving this issue open — Italian is 1 of the 12 languages, and the offer stands for the other 11. And yes, please, on anything else in Italian that reads wrong: the existing translations were not written by native speakers, and you have now demonstrated the exact kind of review that finds those.

**Comment by aavdberg** [comment]
Dutch is complete in PR #483. It translates the four outstanding service blocks in `nl.json`, including the remaining Photo on Demand and session-action field descriptions.

Ukrainian (`uk.json`) is intentionally unchanged for now, pending native-speaker review, so this issue remains open for the other languages.

**Comment by aavdberg** [comment]
Ukrainian is now also complete in PR #483. The four outstanding `uk.json` service blocks were reviewed by a native speaker before inclusion, alongside the Dutch translation already in that PR.

The issue remains open for the other languages.

**Comment by aavdberg** [comment]
Follow-up PR #484 addresses the two Copilot review consistency notes that arrived after #483 had already merged: it standardizes `Alarmpaneelentiteit` and the two `Moet op true staan` confirmation strings. The change is deliberately limited to those three Dutch strings.

**Comment by bvis** [comment]
**Status update.** German, French, European and Brazilian Portuguese and Catalan are in #493, done by me rather than a native speaker, so a word on why after the issue text argued against exactly that.

The argument above was against a maintainer pushing 26 strings through a machine translator and shipping a destructive confirmation he cannot read. What #493 does instead is what wip3out3r did for Italian: every recurring term is lifted from the locale's own file — `Bereich` / `Alarmzentrale`, `espace` / `panneau d'alarme`, `espaço` / `painel de alarme`, `espai` / `panell d'alarma` — so the action reads like the rest of the integration in that language, and the two irreversible `confirm` descriptions are kept deliberately literal, with nothing idiomatic to get wrong. These five are languages I can actually check. **Czech, Polish, Romanian and Turkish are not in it**, on purpose: there I could not vouch for the confirmation wording, so the original reasoning still applies and this issue stays open for them.

Two things that changed the picture while auditing:

- **A fifth, smaller gap nobody had noticed:** the three `press_panic_button` field names — `Entity`, `Latitude`, `Longitude` — were still English in 13 of the 14 locales, *including* `es`, `it`, `nl` and `uk`, whose four blocks were already complete. Only Danish had them localised, because webliodk started from a clean sheet. #493 fixes all 13; in `fr`, `pt` and `pt-BR` only *Entity* changes, since those languages spell the two geographic words the same as English.
- The five new sets follow the Ajax app's own rendering of *Photo on Demand* in each language where the app ships one (`Foto auf Anfrage`, `Photo à la demande`, `Foto a pedido`, `Foto sob demanda`). Catalan has no Ajax app to copy, so `Foto sota demanda` is my choice and the one most worth a native eye.

**What is still wanted here: `cs`, `pl`, `ro`, `tr`** — the four blocks, 26 values each, same instructions as in the description. And for `de`, `fr`, `pt`, `pt-BR`, `ca`: not translation any more, but a native speaker reading the action picker after #493 lands and telling me what reads wrong. Both are equally welcome, as a PR or pasted here.

**Comment by bvis** [comment]
Closing this: the last four languages are in, done here rather than waited for.

`cs`, `pl`, `ro` and `tr` are translated in #530 (merged as ff7de1a) — 104 values across those four files. All 14 locales now carry the four action blocks in their own language, which is the whole of what this issue tracked.

The issue argued against a maintainer translating what he cannot read. That reasoning was not overruled, it was narrowed the same way #477 and #493 narrowed it: nothing is invented where the file already decides it.

- Each of the four files already named the feature somewhere else. `exceptions.photo_no_push` reads `Fotografie na vyžádání`, `Zdjęcie na żądanie`, `Fotografia la cerere`, `İstek üzerine fotoğraf` — so the action now uses the words that locale had already chosen.
- *space*, *alarm panel entity*, *hub*, *account* and *device* come from the `press_panic_button` and `disarm_night_mode` blocks in the same file.
- `Config entry ID` follows Home Assistant core's own translation of that exact field: `ID položky nastavení` (cs), `Yapılandırma girişi kimliği` (tr). Core has nothing for Polish or Romanian there, so Polish borrows core's `ID integracji` and Romanian's `ID intrare de configurare` is a choice with no precedent behind it.
- The two irreversible `confirm` descriptions are kept deliberately literal.

**A native eye is still very welcome — it is now a review rather than a translation.** Three places worth looking at first:

1. **Czech and Romanian render *terminate* as ending a session** (`Ukončit relaci`, `Încheie sesiunea`), where Italian deliberately went the other way in #477 because *terminare una sessione* reads like killing a process. If it reads wrong in yours, say so.
2. **Polish `Jednostka` for the `Entity` field name.** It matches the three sibling actions in `pl.json`, but it means "unit"; `Encja` is core's word and is already what `press_panic_button`'s own description says. The inconsistency is in the file, not in this change, and deserves one pass over the whole file.
3. **Romanian `ID intrare de configurare`** — no precedent anywhere.

A correction is a one-line pull request, or a comment here and I will make it. The same offer stands for anything else that reads badly in your language, including the `de` / `fr` / `pt` / `pt-BR` / `ca` sets from #493, which have still never had a native read.

Strings added to the integration after this sweep are a separate gap and will be tracked on their own rather than here.
